src/js/actions/policy.js

```diff
@@ -204,127 +186,171 @@ define(function (require, exports) {
+            setModeFn = adapterUI.setPointerPropagationMode;
+            break;
+        default:
+            return Promise.reject(new Error("Unknown kind:" + kind));
         }
 
-        var policyStore = this.flux.store("policy");
-        
-        _cachedPolicyState = {
-            keyboardPolicies: policyStore.getPolicies(PolicyStore.eventKind.KEYBOARD),
-            pointerPolicies: policyStore.getPolicies(PolicyStore.eventKind.POINTER)
-        };
-
-        return adapterUI.getKeyboardPropagationMode()
-            .bind(this)
-            .then(function (mode) {
-                _cachedPolicyState.keyboardMode = mode;
-                return adapterUI.getPointerPropagationMode();
-            })
-            .then(function (mode) {
-                _cachedPolicyState.pointerMode = mode;
-                policyStore.clearPolicies(PolicyStore.eventKind.KEYBOARD);
-                policyStore.clearPolicies(PolicyStore.eventKind.POINTER);
-                return Promise.join(
-                    _syncPolicies.call(this, PolicyStore.eventKind.KEYBOARD),
-                    _syncPolicies.call(this, PolicyStore.eventKind.POINTER));
-            })
-            .then(function () {
-                return Promise.join(
-                    adapterUI.setKeyboardPropagationMode(
-                        { defaultMode: adapterUI.keyboardPropagationMode.FOCUS_PROPAGATE }),
-                    adapterUI.setPointerPropagationMode(
-                        { defaultMode: adapterUI.pointerPropagationMode.ALPHA_PROPAGATE }));
-            })
-            .catch(function (err) {
-                _cachedPolicyState = null;
-                throw err;
+        var setModePromise = setModeFn.call(adapterUI, { defaultMode: mode }),
+            dispatchPromise = this.dispatchAsync(events.policies.MODE_CHANGED, {
+                kind: kind,
+                mode: mode
             });
+
+        return Promise.join(setModePromise, dispatchPromise);
     };
-    disablePolicies.reads = [];
-    disablePolicies.writes = [locks.PS_APP, locks.JS_POLICY];
-    disablePolicies.modal = true;
+    setMode.reads = [];
+    setMode.writes = [locks.PS_APP, locks.JS_POLICY];
+    setMode.transfers = [];
+    setMode.modal = true;
 
     /**
-     * Re-enabled cached keyboard policies
+     * Temporarily suspend policies of the given kind
      *
+     * @param {number} kind
      * @return {Promise}
      */
-    var reenablePolicies = function () {
-        if (!_cachedPolicyState) {
-            return Promise.reject(new Error("Can not re-renable keyboard policies, no previously cached state"));
+    var suspendPolicies = function (kind) {
+        var policyStore = this.flux.store("policy");
+        if (policyStore.isSuspended(kind)) {
+            return Promise.reject(new Error("Policies are already suspended"));
         }
+        
+        policyStore.suspend(kind);
+
+        var mode = adapterUI.keyboardPropagationMode.NEVER_PROPAGATE,
```

Comment:
1. It seems a little weird to set the keyboardPropMode inside this generic policy function, since it will get called for any "kind" of policy
2. The default value used to be FOCUS_PROPAGATE, right?  Is this change due to fitz's recent changes in core?  If so, do you know the minimum required build with that?
3. It is also suspicious that you are not somewhere in the `suspend` process setting the default pointer mode ... but maybe that's because its the default value anyway, as you noted in beforeStartup...but anyway I'm not convinced it's not an omission.